tests/neural_networks/test_qcnn.py

```diff
@@ -140,3 +156,39 @@ def test_sequence(self, num_qubits, operations):
                 or mock_quantum_pooling_layer3.call_count > 0
                 or mock_fully_connected_layer.call_count > 0
             )
+
+    # Tests added for BaseLayer inheritance validation
+    @pytest.mark.parametrize(
+        "num_qubits, operations",
+        [
+            (2, [(print, {})]),
+            (3, [(sum, {})]),
+            (4, [(lambda x: x, {})]),
+        ],
+    )
+    def test_non_class_operations(self, num_qubits, operations):
+        """Tests that non-class callables are rejected."""
+        with raises(TypeError, match="must be a class"):
+            _ = QCNN(num_qubits).sequence(operations)
+
+    @pytest.mark.parametrize(
+        "num_qubits, operations",
+        [
+            (2, [(dict, {})]),
+            (3, [(str, {})]),
+            (4, [(object, {})]),
```

Comment:
You can also add parameterizations with other classes from inside the library such as as, `MERA`, `QCNN`, etc. to verify if the test case catches them as well.